test/types/querycursor.test.ts

```diff
@@ -20,3 +20,30 @@ Test.find().cursor().
     expectType<number>(i);
   }).
   then(() => console.log('Done!'));
+
+async function gh14374() {
+  // `Parent` represents the object as it is stored in MongoDB
+  interface Parent {
+    child?: Types.ObjectId
+    name?: string
+  }
+  const ParentModel = model<Parent>(
+    'Parent',
+    new Schema({
+      child: { type: Schema.Types.ObjectId, ref: 'Child' },
+      name: String
+    })
+  );
+
+  interface Child {
+    name: string
+  }
+  const childSchema: Schema = new Schema({ name: String });
+
+  // This does not
+  const cursor = ParentModel.find({}).populate<{ child: Child }>('child').cursor();
+  for await (const doc of cursor) {
+    const _t: string = doc.child.name;
```

Comment:
maybe add a `expectType` here for `doc`, or is it a very complex type here?

Reply:
`doc` is a bit complex, but I added an `expectType` on `doc.child`.